driver-core/src/main/com/mongodb/internal/session/ServerSessionPool.java

```diff
@@ -43,23 +41,20 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static com.mongodb.assertions.Assertions.isTrue;
-import static com.mongodb.internal.connection.ConcurrentPool.INFINITE_SIZE;
 import static java.util.concurrent.TimeUnit.MINUTES;
 
 public class ServerSessionPool {
-    private static final int END_SESSIONS_BATCH_SIZE = 10000;
-
-    private final ConcurrentPool<ServerSessionImpl> serverSessionPool =
-            new ConcurrentPool<>(INFINITE_SIZE, new ServerSessionItemFactory());
+    private final ConcurrentLinkedDeque<ServerSessionImpl> available = new ConcurrentLinkedDeque<>();
     private final Cluster cluster;
     private final ServerSessionPool.Clock clock;
-    private volatile boolean closing;
     private volatile boolean closed;
-    private final List<BsonDocument> closedSessionIdentifiers = new ArrayList<>();
     @Nullable
     private final ServerApi serverApi;
+    private final AtomicInteger inUseCount = new AtomicInteger();
```

Comment:
ConcurrentPool used to do this for us.

Reply:
Yes.  But not just for unit tests, also for the assertion we make after every integration test to insure that no sessions are leaked.

Reply:
Done